### PR TITLE
Implemented the Admin user story 3, 4, 5, 6, 7

### PR DESCRIPTION
**Pull request by rommeltorquator**
#### Implemented the Admin user story 3, 4, 5, 6, 7

[WIP] Broker confirmation email
[WIP] Buyer success registration email
[WIP] Account creation using the Admin account

## The Story
This is the first week since the start of the project. I am also relieved that we were able to implement almost every user stories for this project. Jaren is also currently implementing the Rspec test for our app, which is a very huge task, but I have tremendous faith that he will deliver. With the current pace that we have, I believe that we will be able to deliver the MVP on time or ahead of schedule. Here is what I did in the PR.
- Created a transactions page on admin account to see all transactions of every user of the app.
- Built a dashboard page for admin account to show the number of current buyers, brokers and transaction, and pending approvals, list of users.
- Enabled the admin account to approve a pending broker account request.
- Created a show page for individual user.

#### Admin dashboard page
image
#### User show page
image
#### Transactions page
image

## Resources
https://juliannaroen.medium.com/rails-callbacks-cheat-sheet-824295a1a14d

### CODE REVIEW

**Review by analyn-cajocson — CHANGES_REQUESTED** [review]
Kindly rebase your branch against the updated `master`

**Comment by rommeltorquator** [comment]
All the merge conflicts have been successfully resolved.

File: app/controllers/home_controller.rb [hunk 1]
```diff
@@ -3,12 +3,31 @@ def index
...
+    @pending_approval = User.where(approved: false)
+    @transactions = PurchaseTransaction.all
   end
 
   def portfolio; end
 
   def transaction
     @transactions = current_buyer.purchase_transactions if current_buyer
     @transactions = PurchaseTransaction.where(broker_id: current_broker.id) if current_broker
+    @transactions = PurchaseTransaction.all if current_admin
```
Comment (analyn-cajocson) [inline app/controllers/home_controller.rb:15]:
```suggestion
    @transactions = if current_buyer
                  current_buyer.purchase_transactions
                elsif current_broker
                  PurchaseTransaction.where(broker_id: current_broker.id)
                elsif current_admin
                  PurchaseTransaction.all
                end
```

File: app/models/broker.rb [hunk 2]
```diff
@@ -1,2 +1,9 @@
 class Broker < User
+  before_create :unapprove_broker
```
Comment (analyn-cajocson) [inline app/models/broker.rb:2]: I think you can use the `before_validation` ?

File: app/views/home/show_user.html.erb [hunk 3]
```diff
@@ -0,0 +1,10 @@
+<h1>User show page</h1>
+<p>first name: <%= @user.first_name %></p>
+<p>last name: <%= @user.last_name %></p>
+<p>email: <%= @user.email %></p>
+User type:
+<% if @user.type == 'Broker' %>
+  <strong>Broker</strong>
+<% else %>
+  <strong>Buyer</strong>
+<% end %>
```
Comment (analyn-cajocson) [inline app/views/home/show_user.html.erb:10]: add new line

File: app/views/home/transaction.html.erb [hunk 4]
```diff
@@ -19,4 +23,4 @@
       </div>
     <% end %>
   </div>
-</div>
+</div>
```
Comment (analyn-cajocson) [inline app/views/home/transaction.html.erb:26]: add new line

**Review by analyn-cajocson — CHANGES_REQUESTED** [review]

**Pull request closed without merging**